pkg/cache/internal/cache_reader.go

```diff
@@ -150,8 +165,15 @@ func (c *CacheReader) List(_ context.Context, out client.ObjectList, opts ...cli
 			}
 		}
 
-		outObj := obj.DeepCopyObject()
-		outObj.GetObjectKind().SetGroupVersionKind(c.groupVersionKind)
+		var outObj runtime.Object
+		if c.disableDeepCopy != nil {
+			// skip deep copy which might be unsafe
+			// you must DeepCopy any object before mutating it outside
+			outObj = obj
```

Comment:
Do we need to set the GVK here?

Reply:
Set the GVK here might modify the object in Informer. We may just ignore it according to the [discussion earlier](https://github.com/kubernetes-sigs/controller-runtime/pull/1274#discussion_r678615164).

Reply:
Ah got it! That earlier discussion makes more sense to me now :)